sys/exec.c

```diff
@@ -242,6 +230,19 @@ int do_exec(const exec_args_t *args) {
     proc_populate(proc, td);
   }
 
+  /* Prepare file descriptor table */
+  /* TODO: Copy/share file descriptor table! */
```

Comment:
You are correct. Initially `do_exec` was not meant to serve a system call (back then we did not have working syscalls), but I'm starting to move it into that direction. Thus I suppose the file descriptor table should only be created when starting the very first process (that is: `do_exec` was called by a kernel thread), otherwise the table should stay unmodified and it is `fork` who should create a copy or shared reference. I'll fix that ASAP.